src/pydocstyle/checker.py

```diff
@@ -997,9 +997,23 @@ def get_leading_words(line):
         return result.group()
 
 
-def get_function_args(function_string):
+def get_function_args(function_definition):
     """Return the function arguments given the source-code string."""
-    function_arg_node = ast.parse(textwrap.dedent(function_string)).body[0].args
```

Comment:
I think it may be much easier to just remove leading indentation from the first line of `function_string`, and not even bother dedenting the rest?  i.e. just replace `ast.parse(textwrap.dedent(function_string))` by `ast.parse(function_string.lstrip())` and not touch anything else.

Reply:
Tried this. Doesn't work

Reply:
My bad - this ended up being a cleaner solution! Thanks for the idea.